packages/services/src/features/localStorage.service.ts

```diff
@@ -22,20 +22,65 @@ export const saveToLocalStorage = (data: any, parameters: { storageKey: string }
  */
 export const mergeToLocalStorage = (
   data: any,
-  parameters: { storageKey: string; objectKeyPath: string; objectValuePath?: string },
+  parameters: { storageKey: string; objectKeyPath?: string; objectValuePath?: string },
 ) => {
   try {
-    const { storageKey, objectKeyPath } = parameters;
+    const { storageKey } = parameters;
     const existingData = localStorage.getItem(storageKey);
-    const key = getValueByPath(data, objectKeyPath);
     const value = parameters.objectValuePath ? getValueByPath(data, parameters.objectValuePath) : data;
 
     if (existingData) {
       const parsedData = JSON.parse(existingData);
-      localStorage.setItem(storageKey, JSON.stringify({ ...parsedData, [key]: value }));
+      if (parameters.objectKeyPath) {
+        const key = getValueByPath(data, parameters.objectKeyPath);
+
+        localStorage.setItem(storageKey, JSON.stringify({ ...parsedData, [key]: value }));
+      } else {
+        localStorage.setItem(storageKey, JSON.stringify({ ...parsedData, ...value }));
+      }
     } else {
-      localStorage.setItem(storageKey, JSON.stringify({ [key]: value }));
+      if (parameters.objectKeyPath) {
+        const key = getValueByPath(data, parameters.objectKeyPath);
+
+        localStorage.setItem(storageKey, JSON.stringify({ [key]: value }));
+      } else {
+        localStorage.setItem(storageKey, JSON.stringify({ ...value }));
+      }
+    }
+  } catch (error: any) {
+    console.error(error);
+    throw new Error(error.message);
+  }
+};
+
+/**
+ * This function is used to get the data from the local storage
+ * @param parameters.storageKey - The key of the local storage item.
+ * @param parameters.key - The key to be retrieved from the local storage item. If not provided, the entire item is returned.
+ * @returns data
+ */
+export const getValueFromLocalStorage = (
+  data: any,
+  parameters: { storageKey: string; key?: string; stateKey?: string },
+) => {
+  try {
+    const _data = { ...data };
+    const { storageKey, key } = parameters;
+    const existingData = localStorage.getItem(storageKey);
+    let retrievedData;
+    if (existingData) {
+      const parsedData = JSON.parse(existingData);
+      if (!key) {
+        retrievedData = parsedData;
+      } else {
+        retrievedData = parsedData[key];
+      }
+    }
+    if (parameters.stateKey) {
+      _data[parameters.stateKey] = retrievedData;
+      return _data;
     }
+    return retrievedData;
   } catch (error: any) {
     console.error(error);
     throw new Error(error.message);
```

Comment:
Could you write unit tests for this function to ensure that it behaves as expected?

Reply:
Thanks @huynguyen-hl , I updated it